src/plugins.ts

```diff
@@ -0,0 +1,16 @@
+type BuilderFacade = {
+  // addType: (def: any) => void;
+  hasType: (typeName: string) => boolean;
+};
+
+type Plugin = (hooks: Hooks) => void;
+
+type Hooks = {
+  onInstall: OnInstallHook;
+};
+
+type OnInstallHook = (
+  onInstallHandler: (builder: BuilderFacade) => { types: any[] }
+) => void;
+
+export { OnInstallHook, Plugin, Hooks };
```

Comment:
How do you feel about batch export at end of file? I like the symmetry with batch import at beginning of file. Also less clutter I think. Colocating the answer to "is this exported" is lost though. Arguably IDE outline view etc. can support but not enough I guess.